### PR TITLE
Fix: `eslint:all` causes regression in 3.1.0 (fixes #6687)

### LINKED ISSUE #6687: eslint:all causes regression in v3.1.0
**What version of ESLint are you using?**

v3.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

<details>
``` json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "jsx": true,
      "impliedStrict": true,
      "experimentalObjectRestSpread": true
    },
    "sourceType": "module"
  },
  "env": {
    "browser": true,
    "node": true,
    "jest": true,
    "es6": true
  },
  "extends": [
    "eslint:all",
    "plugin:jasmine/recommended"
  ],
  "parser": "babel-eslint",
  "rules": {
    "quotes": ["error", "single"],
    "strict": ["error", "never"],
    "array-bracket-spacing": ["error", "never"],
    "block-scoped-var": "warn",
    "camelcase": "warn",
    "comma-spacing": ["error", {"before": false, "after": true}],
    "computed-property-spacing": ["error", "never"],
    "complexity": ["error", 11],
    "consistent-return": "error",
    "constructor-super": "error",
    "no-duplicate-imports": "error",
    "curly": "warn",
    "eqeqeq": "error",
    "eol-last": "error",
    "indent": ["error", 2, {"VariableDeclarator": 2}],
    "max-depth": ["error", 4],
    "no-alert": "warn",
    "no-array-constructor": "error",
    "no-bitwise": "error",
    "no-caller": "error",
    "no-unsafe-finally": "error",
    "no-useless-computed-key": "error",
    "no-catch-shadow": "error",
    "no-continue": "error",
    "no-cond-assign": "error",
    "no-div-regex": "error",
    "no-else-return": "error",
    "no-eq-null": "error",
    "no-extend-native": "error",
    "no-floating-decimal": "error",
    "no-implied-eval": "error",
    "no-inline-comments": "error",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "error",
    "no-multiple-empty-lines": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-require": "error",
    "no-new-wrappers": "error",
    "no-param-reassign": "error",
    "no-path-concat": "error",
    "no-const-assign": "error",
    "no-dupe-class-members": "error",
    "no-eval": "error",
    "no-extra-bind": "error",
    "no-shadow": "error",
    "no-multi-spaces": "warn",
    "no-native-reassign": "error",
    "no-nested-ternary": "warn",
    "no-proto": "error",
    "no-throw-literal": "error",
    "no-redeclare": "error",
    "no-undef": "error",
    "no-use-before-define": "error",
    "no-unused-vars": "error",
    "prefer-spread": "warn",
    "prefer-arrow-callback": "warn",
    "arrow-spacing": "warn",
    "prefer-const": "warn",
    "prefer-template": "warn",
    "no-var": "error",
    "no-useless-escape": "error",
    "max-len": ["warn", 120],
    "max-statements-per-line": ["error", { "max": 3 }],
    "jsx-quotes": "warn",
    "keyword-spacing": "error",
    "promise/param-names": "warn",
    "promise/always-return": "warn",
    "promise/catch-or-return": "warn",
    "import-order/import-order": "error",
    "babel/generator-star-spacing": "warn",
    "babel/new-cap": "warn",
    "babel/object-shorthand": "warn",
    "babel/arrow-parens": "warn",
    "babel/no-await-in-loop": "warn",
    "babel/object-curly-spacing": "off",
    "babel/array-bracket-spacing": "error",
    "babel/flow-object-type": ["error", "comma"],
    "immutable/no-let": "warn",
    "jasmine/no-spec-dupes": [1, "branch"],
    "react/display-name": "warn",
    "react/forbid-prop-types": "warn",
    "react/jsx-boolean-value": ["warn", "always"],
    "react/jsx-closing-bracket-location": "warn",
    "react/jsx-curly-spacing": "warn",
    "react/jsx-equals-spacing": ["error", "never"],
    "react/jsx-first-prop-new-line": ["error", "multiline"],
    "react/jsx-handler-names": "warn",
    "react/jsx-indent-props": ["warn", 2],
    "react/jsx-indent": ["warn", 2],
    "react/jsx-key": "warn",
    "react/jsx-max-props-per-line": ["warn", { "maximum": 2}],
    "react/jsx-no-bind": ["error"],
    "react/jsx-no-duplicate-props": "warn",
    "react/jsx-no-literals": "off",
    "react/jsx-no-undef": "error",
    "react/jsx-no-target-blank": "error",
    "react/jsx-pascal-case": "warn",
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "react/jsx-sort-props": ["warn", {
      "callbacksLast": true,
      "shorthandFirst": true,
      "ignoreCase": true
    }],
    "react/jsx-sort-prop-types": "off",
    "react/jsx-space-before-closing": "off",
    "react/no-danger": "warn",
    "react/no-deprecated": "warn",
    "react/no-did-mount-set-state": "warn",
    "react/no-did-update-set-state": "warn",
    "react/no-direct-mutation-state": "warn",
    "react/no-is-mounted": "warn",
    "react/no-multi-comp": "warn",
    "react/no-string-refs": "warn",
    "react/no-comment-textnodes": "error",
    "react/no-unknown-property": "error",
    "react/prefer-es6-class": "warn",
    "react/prop-types": "error",
    "react/react-in-jsx-scope": "error",
    "react/require-extension": "warn",
    "react/require-render-return": "error",
    "react/self-closing-comp": "warn",
    "react/sort-comp": "error",
    "react/sort-prop-types": ["warn", {
      "callbacksLast": true,
      "ignoreCase": true,
      "requiredFirst": true
    }],
    "react/wrap-multilines": "warn",
    "flowtype/space-after-type-colon": [1, "always"],
    "flowtype/space-before-type-colon": [1, "never"],
    "flowtype/type-id-match": [1, "^([A-Z][a-z0-9]+)"],
    "flowtype/require-valid-file-annotation": ["error", "always"],
    "semi": ["error", "never"],
    "flow-vars/define-flow-type": "warn",
    "flow-vars/use-flow-type": "warn",
    "array-callback-return": "off",
    "arrow-body-style": "off",
    "block-spacing": "off",
    "brace-style": "off",
    "dot-location": "off",
    "dot-notation": "off",
    "func-names": "off",
    "func-style": "off",
    "generator-star-spacing": "off",
    "global-require": "off",
    "id-length": "off",
    "init-declarations": "off",
    "key-spacing": "off",
    "lines-around-comment": "off",
    "max-lines": "off",
    "max-params": "off",
    "max-statements": "off",
    "new-cap": "off",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "no-empty-function": "off",
    "no-extra-parens": "off",
    "no-implicit-coercion": "off",
    "no-invalid-this": "off",
    "no-magic-numbers": "off",
    "no-mixed-operators": "off",
    "no-negated-condition": "off",
    "no-plusplus": "off",
    "no-process-env": "off",
    "no-sequences": "off",
    "no-spaced-func": "off",
    "no-sync": "off",
    "no-ternary": "off",
    "no-trailing-spaces": "off",
    "no-undefined": "off",
    "no-underscore-dangle": "off",
    "no-unneeded-ternary": "off",
    "no-unused-expressions": "off",
    "object-curly-newline": ["off"],
    "object-curly-spacing": ["off"],
    "object-property-newline": "off",
    "one-var": ["error", "never"],
    "operator-assignment": "off",
    "operator-linebreak": "off",
    "padded-blocks": "off",
    "prefer-reflect": "off",
    "quote-props": "off",
    "require-jsdoc": "off",
    "sort-imports": "off",
    "space-before-blocks": "off",
    "space-before-function-paren": "off",
    "space-in-parens": "off",
    "space-infix-ops": "off",
    "space-unary-ops": "off",
    "spaced-comment": "off",
    "valid-jsdoc": "off",
    "wrap-regex": "off",
    "multiline-ternary": "off"
  },
  "plugins": [
    "babel",
    "flow-vars",
    "flowtype",
    "immutable",
    "import-order",
    "jasmine",
    "promise",
    "react"
  ]
}
```
</details>

**What did you do? Please include the actual source code causing the issue.**

ran `eslint .` using the configuration above.

**What did you expect to happen?**

Eslint runs as usual.

**What actually happened? Please include the actual, raw output from ESLint.**

For every file in the project, I got the following error:

`2:1  error  Definition for rule '.eslintrc.yml' was not found  .eslintrc.yml`

### PR DESCRIPTION
**Pull request by alberto**
**What issue does this pull request address?**
#6687 

**What changes did you make? (Give an overview)**
Filter files by `.js` extension

**Is there anything you'd like reviewers to focus on?**
No.

This replaces #6685 since the fix wasn't correct, and we might want to include this in a patch release tomorrow.

### CODE REVIEW

File: conf/eslint-all.js [hunk 1]
```diff
@@ -16,7 +16,11 @@ var fs = require("fs"),
 // Helpers
 //------------------------------------------------------------------------------
 
-var ruleFiles = fs.readdirSync(path.resolve(__dirname, "../lib/rules"));
+var ruleFiles = fs.readdirSync(
+        path.resolve(__dirname, "../lib/rules")
+    ).filter(function(file) {
```
Comment (ilyavolodin) [inline conf/eslint-all.js]: Do we need both filter and reduce? Could we just replace them both with a single forEach loop instead?

Comment (alberto) [inline conf/eslint-all.js]: I kept it separate to make it cleaner, but it could also be part of the reduce.

**Comment by alberto** [comment]
I changed it to a single pass and added a couple of tests to get some certainty it's returning all rules.

**Comment by ilyavolodin** [comment]
LGTM. Thanks

**Comment by nzakas** [comment]
Lgtm. I know this has only been open < 1 day, but seems like we should make an exception and merge this for today's patch release? @eslint/eslint-tsc

**Comment by ilyavolodin** [comment]
Agree. This is simple enough change, it should not create any kind of problems. And we want to do a patch release to fix this as soon as possible.

**Pull request merged**